tests/testthat/test-shinytest2-landing_popup.R

```diff
@@ -0,0 +1,184 @@
+testthat::test_that("e2e: teal app with landing_popup_module initializes with no errors", {
+  app <- TealAppDriver$new(
+    data = simple_teal_data(),
+    modules = modules(
+      landing_popup_module(
+        title = "Welcome",
+        content = tags$b("A welcome message!", style = "color: red;")
+      ),
+      example_module()
+    )
+  )
+  app$view()
+
+  app$wait_for_idle(timeout = default_idle_timeout)
+  app$expect_no_shiny_error()
+  app$stop()
+})
+
+test_that("e2e: app with default landing_popup_module creates modal containing a button", {
+  app <- TealAppDriver$new(
+    data = simple_teal_data(),
+    modules = modules(
+      landing_popup_module(),
+      example_module()
+    )
+  )
+  app$wait_for_idle(timeout = default_idle_timeout)
+
+  expect_equal(
+    app$get_html(".btn-default") %>%
+      grep("Accept", x = ., value = TRUE) %>%
+      rvest::read_html() %>%
+      rvest::html_text(),
+    "Accept"
+  )
+
+  app$stop()
+})
+
+test_that("e2e: when default landing_popup_module is closed, it shows the underlying teal app", {
+  app <- TealAppDriver$new(
+    data = simple_teal_data(),
+    modules = modules(
+      landing_popup_module(),
+      example_module()
+    )
+  )
+  app$wait_for_idle(timeout = default_idle_timeout)
+
+  app$click(selector = ".btn-default:nth-child(1)")
+
+  # QUESTION: Is there a better way of checking the name of current module?
+  expect_equal(
+    app$get_html("#teal-main_ui-root-active_tab") %>%
+      rvest::read_html() %>%
+      rvest::html_node("a") %>%
+      rvest::html_attr("data-value"),
+    "example_teal_data"
+  )
+
+  app$stop()
+})
+
+
+# customized landing_popup_module ---------------------------------------------------------------------------------
+
+
+extract_onclick <- function(id) {
+  app$get_html(id) %>%
+    rvest::read_html() %>%
+    rvest::html_nodes("button") %>%
+    rvest::html_attr("onclick")
+}
+extract_text <- function(id) {
+  app$get_html(id) %>%
+    rvest::read_html() %>%
+    rvest::html_text() %>%
```

Comment:
I believe this is equivalent, but the tests below are failing and can't confirm :sweat_smile: 

```suggestion
  app$get_text(id) %>%
```

Reply:
Will double check!

Reply:
one of the tests needed `trimws()` after applying `app$get_text()`